types/twitch-ext/index.d.ts

```diff
@@ -170,6 +176,44 @@ interface TwitchExtActions {
 	requestIdShare(): void;
 }
 
+/**
+ * @see TwitchExt.configuration
+ */
+interface TwitchExtConfiguration {
+	/**
+	 * This property returns the record for the broadcaster segment if one is found; otherwise, undefined.
+	 */
+	broadcaster: { version: string; content: string } | undefined;
+
+	/**
+	 * This property returns the record for the developer segment if one is found; otherwise, undefined.
+	 */
+	developer: { version: string; content: string } | undefined;
+
+	/**
+	 * This property returns the record for the global segment if one is found; otherwise, undefined.
+	 */
+	global: { version: string; content: string } | undefined;
+
+	/**
+	 * This function registers a callback that is called whenever an extension configuration is received.
+	 * The callback function takes no input and returns nothing. After this is called for the first time,
+	 * the records for the global, developer and broadcaster segments will be set if the data is available.
+	 * @param callback The callback that is fired.
+	 */
+	onChanged(
+		callback: () => void
+	): void;
+
+	/**
+	 * This function can be called by the front end to set an extension configuration.
+	 * @param segment The string-encoded configuration.
+	 * @param version The configuration segment to set.
+	 * @param content The version of configuration with which the segment is stored.
+	 */
+	set(segment: string, version: string, content: string): void;
```

Comment:
```suggestion
	set(segment: "broadcaster", version: string, content: string): void;
```
Let's enforce `"broadcaster"` here, as that's the [only valid value](https://dev.twitch.tv/docs/extensions/reference/#helper-configuration), and it's easy enough to mess up the order otherwise.